internal/services/order/order.go

```diff
@@ -1,13 +1,219 @@
 package order
 
-func BuyOrder(BReq BuyRequest) (bRes BuyResponse, err error) {
-	return bRes, err
+import (
+	"Tradeasy/config"
+	"Tradeasy/internal/model"
+	"Tradeasy/internal/services/stock_exchange"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"github.com/google/uuid"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type result struct {
+	StockName     string
+	TotalQuantity int
 }
 
-func SellOrder(BReq SellRequest) (sRes SellResponse, err error) {
-	return sRes, err
+func BuyOrder(bReq BuyRequest) (bRes stock_exchange.OrderResponse, err error) {
+	var stocks model.StocksFeed
+	var account model.TradingAccount
+
+	if bReq.BookType == market {
+		if err = config.DB.Table("stocks_feed").Where("stock_name=?", bReq.StockName).Last(&stocks).Error; err != nil {
+			return bRes, err
+		}
+		bReq.LimitPrice = stocks.LTP
+	}
+
+	if err = config.DB.Table("trading_account").Where("user_id=?", bReq.UserId).First(&account).Error; err != nil {
+		return bRes, err
+	}
+
+	if account.Balance < int64(bReq.Quantity*bReq.LimitPrice) {
+		return bRes, errors.New("balance is insufficient for the placed order")
+	}
+
+	account.Balance = account.Balance - int64(bReq.Quantity*bReq.LimitPrice)
+	//account.UpdatedAt=time.Now()
+	if err = config.DB.Table("trading_account").Where("user_id=?", bReq.UserId).Updates(&account).Error; err != nil {
+		return bRes, err
+	}
+
+	orderId := uuid.New().String()
+	p := model.PendingOrders{
+		UserId:    bReq.UserId,
+		OrderId:   orderId,
+		StockName: bReq.StockName,
+		OrderType: buy,
+		BookType:  bReq.BookType,
+		Quantity:  bReq.Quantity,
+		Status:    pending,
+	}
+
+	if bReq.BookType == market {
+		p.OrderPrice = bReq.LimitPrice
+		if err = config.DB.Table("pending_orders").Create(&p).Error; err != nil {
+			return bRes, err
+		}
+	} else {
+		p.LimitPrice = bReq.LimitPrice
+		if err = config.DB.Table("pending_orders").Create(&p).Error; err != nil {
+			return bRes, err
+		}
+	}
+
+	exeOrder := stock_exchange.OrderRequest{
+		OrderID:    orderId,
+		StockName:  bReq.StockName,
+		Quantity:   bReq.Quantity,
+		OrderType:  bReq.BookType,
+		LimitPrice: bReq.LimitPrice,
+	}
+
+	request, err := json.Marshal(exeOrder)
+	if err != nil {
+		return bRes, err
+	}
+	response, err := http.Post(buyOrderURL, "application/json", bytes.NewBuffer(request))
+	if err != nil {
```

Comment:
Order is in pending even if exchange was not able to place your order. Move it to pending only if exchange accept your order.

Reply:
Done..moved the create part after checking